### PR TITLE
Use bot registry in workflow step

### PR DESCRIPTION
**Pull request by psrpinto**
### Summary of changes:

- Add `GetPrimaryClient()`to bot registry, which retrieves a matrix client authenticated with the "god" bot (from now on knonw as "primary bot" :) )
- Use bot registry in workflow step
- Remove intances of engine where they're no longer needed

### CODE REVIEW

File: engine/workflow_step_matrix_post_message.go [hunk 1]
```diff
@@ -32,25 +32,12 @@ var getMatrixClient = func(homeserver string) (MatrixClient, error) {
...
-		if !b.IsHydrated() {
-			b.Hydrate(e)
-		}
-
-		return b.getMCInstance(), nil
+		return s.botRegistry.GetPrimaryClient()
 	}
 
-	return e.client, nil
+	return s.botRegistry.GetClient(s.asBot)
```
Comment (ashfame) [inline engine/workflow_step_matrix_post_message.go:40]: @psrpinto We did this wrong. Inside the `if` it should be a non primary bot and outside (default) it should be the primary client

**Review by ashfame — CHANGES_REQUESTED** [review]
Looks good! This also needs resolving conflicts now that I have merged my #72

I would prefer we merge master in this branch to resolve conflicts and then merge the PR in the usual way. That's the cleanest approach IMO. Happy to discuss this further if you think otherwise ✌️

File: engine/workflow_step_matrix_post_message.go (hunk 1 above)
Comment (psrpinto) [reply engine/workflow_step_matrix_post_message.go:40]: Great catch! Fixing.

**Review by ashfame — APPROVED** [review]
🚀

**Pull request merged**